### PR TITLE
Fix #1931. Implemented full screen button

### LINKED ISSUE #1931: Geometric operation DWITHIN and WITHIN returns wrong results
how to reproduce:
- Open the Test WFS Map
- Create a polygon as descrivbed in the picture below
- Click search

image
It should return only WA and OR but it seems to use degrees and not meters

here is the request
```
curl 'https://demo.geo-solutions.it/geoserver/wfs?service=WFS&outputFormat=json' -H 'Origin: http://localhost:8081' -H 'Accept-Encoding: gzip, deflate, br' -H 'Accept-Language: it-IT,it;q=0.8,en-US;q=0.6,en;q=0.4' -H 'User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.86 Safari/537.36' -H 'Content-Type: application/json' -H 'Accept: application/json' -H 'Referer: http://localhost:8081/?debug=true' -H 'Connection: keep-alive' --data-binary '<wfs:GetFeature startIndex="0" maxFeatures="20" service="WFS" version="1.1.0" xmlns:gml="http://www.opengis.net/gml" xmlns:wfs="http://www.opengis.net/wfs" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.opengis.net/wfs http://schemas.opengis.net/wfs/1.1.0/wfs.xsd"><wfs:Query typeName="topp:states" srsName="EPSG:900913"><ogc:Filter><ogc:DWithin><ogc:PropertyName>the_geom</ogc:PropertyName><gml:Polygon srsName="EPSG:900913"><gml:exterior><gml:LinearRing><gml:posList>-13912762.14035465 6447616.209911188 -14225848.20821073 5811660.134578522 -13589892.132878065 5410518.610137917 -13306157.883883491 6046474.685470584 -13912762.14035465 6447616.209911188</gml:posList></gml:LinearRing></gml:exterior></gml:Polygon><ogc:Distance units="m">10</ogc:Distance></ogc:DWithin></ogc:Filter></wfs:Query></wfs:GetFeature>' --compressed
```

### PR DESCRIPTION
**Pull request by offtherailz**
- Fix #1931. Full screen button added to desktop, embedded and mobile versions
 - Now all modals and dialogs work in embedded and fullscreen mode
 - Review of zoomto icon (too similar to full screen)
 - Added tutorial step for fullscreen button
image

### CODE REVIEW

File: web/client/components/TOC/fragments/SettingsModal.jsx [hunk 1]
```diff
@@ -153,7 +153,7 @@ const SettingsModal = React.createClass({
 
         if (this.props.settings.expanded) {
             return this.props.asModal ? (
-                <Modal {...this.props.options} show={this.props.settings.expanded} container={document.getElementById("body")}>
+                <Modal {...this.props.options} container={document.querySelector(".ms2 > #container") } show={this.props.settings.expanded}>
```
Comment (mbarto) [inline web/client/components/TOC/fragments/SettingsModal.jsx]: cannot be fixed, at least it should be a property, better if it's not spread into every dialog. We should find a way to make it transparent. ms2 is overridable. #container is worse, this is a container, we cannot know the name, and we don't want to know it

File: web/client/components/TOC/fragments/SettingsModal.jsx [hunk 2]
```diff
@@ -162,7 +162,7 @@ const SettingsModal = React.createClass({
                         {footer}
                     </Modal.Footer>
                 </Modal>
-            ) : (<Portal><Dialog id={this.props.id} style={this.props.panelStyle} className={this.props.panelClassName}>
+            ) : (<Portal container={document.querySelector(".ms2 > #container") }><Dialog id={this.props.id} style={this.props.panelStyle} className={this.props.panelClassName}>
```
Comment (mbarto) [inline web/client/components/TOC/fragments/SettingsModal.jsx]: Same here

File: web/client/components/buttons/InfoButton.jsx [hunk 3]
```diff
@@ -86,6 +86,7 @@ const InfoButton = React.createClass({
     },
     render() {
         const dialog = this.props.useModal ? (<Modal
+            container={document.querySelector(".ms2 > #container") }
```
Comment (mbarto) [inline web/client/components/buttons/InfoButton.jsx]: Same here

File: web/client/components/buttons/__tests__/FullScreenButton-test.jsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2015, GeoSolutions Sas.
+ * All rights reserved.
```
Comment (MV88) [inline web/client/components/buttons/__tests__/FullScreenButton-test.jsx]: 2015 to 2017

File: web/client/plugins/FullScreen.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016, GeoSolutions Sas.
+ * All rights reserved.
```
Comment (MV88) [inline web/client/plugins/FullScreen.jsx]: same here

File: web/client/epics/fullscreen.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016, GeoSolutions Sas.
+ * All rights reserved.
```
Comment (MV88) [inline web/client/epics/fullscreen.js]: same here --> 2017

**Pull request merged**